spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala

```diff
@@ -138,6 +139,14 @@ class BigQueryStrategy(expressionConverter: SparkExpressionConverter, expression
             case Sort(orderExpr, true, _) =>
               SortLimitQuery(expressionConverter, expressionFactory, None, orderExpr, subQuery, alias.next)
 
+            case Window(windowExpressions, _, _, _) =>
+              /**
+               * Need to send in the query attributes along with window expressions to avoid "TreeNodeException: Binding attribute, tree"
+               */
+              val projections: Seq[NamedExpression] =
```

Comment:
So, instead of passing projections from here, why don't we create it in the WindowQuery class itself? We can create a WindowQuery companion object and add a method there

Reply:
Handled in the WindowQuery class itself